Домашняя работа с алертами. Выполнено:

dimov-danya1999 opened the pull request:
1. При попадании на этот экран запрашивать ФИО через алер и отображает в UILabel
2. По нажатию на "сложение" открывать Алерт который просит ввести число 1 и число 2 и кнопка посчитать выводит результат в любом виде
3. При нажатии на кнопку "угадай число" - появляеется рандомное число
4. Создана кнопка "начать" которая вызывает Алерт с сообщением "введите слово"
5. При вводе "leohl" в Алерт  мы отдаем контролеру а контролер отдаст это в модель которая из буквосочетания "leohl" вернет слово "hello" контролеру а контролер отдаст это лейблу и отобразит это на экране

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+
+final class MainViewController: UIViewController {
+    
+    private var model = FIOModel()
```
****
```suggestion
    private var user = User()
```

dimov-danya1999 replied:
Исправил

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+        return randomLabel
+    }()
+    
+    private lazy var nameLabel: UILabel = {
```
private  let nameLabel = UILabel()
private  let randomLabel = UILabel()

//код пишется во viewDidLoad

[nameLabel, randomLabel].forEach { 
$0.translatesAutoresizingMaskIntoConstraints = false
}

dimov-danya1999 replied:
Исправил

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+        return addSumButton
+    }()
+    
+    private lazy var guessNumberButton: UIButton = {
```
Вынести в отдельный UI элемент

dimov-danya1999 replied:
Исправил

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+        setupLayout()
+    }
+
+    @objc func validityHello() {
```
private

dimov-danya1999 replied:
Исправил

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+
+    @objc func validityHello() {
+        let alert = UIAlertController(title: "Верни hello", message: "Введи leohl", preferredStyle: .alert)
+        let action = UIAlertAction(title: "Вернуть", style: .default) { (action ) in
```
```suggestion
        let action = UIAlertAction(title: "Вернуть", style: .default) { action in
```

dimov-danya1999 replied:
Исправил

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+    @objc func validityHello() {
+        let alert = UIAlertController(title: "Верни hello", message: "Введи leohl", preferredStyle: .alert)
+        let action = UIAlertAction(title: "Вернуть", style: .default) { (action ) in
+            self.helloLabel.text = self.helloTextModel.textHello(alert.textFields?.first?.text ?? "")
```
переделать

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+        let alert = UIAlertController(title: "Верни hello", message: "Введи leohl", preferredStyle: .alert)
+        let action = UIAlertAction(title: "Вернуть", style: .default) { (action ) in
+            self.helloLabel.text = self.helloTextModel.textHello(alert.textFields?.first?.text ?? "")
+           
```
лишний enter

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+            self.helloLabel.text = self.helloTextModel.textHello(alert.textFields?.first?.text ?? "")
+           
+        }
+        alert.addTextField{ (textField: UITextField) in
```
```suggestion
        alert.addTextField { textField in
```

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+            textField.keyboardType = .default
+        }
+        alert.addAction(action)
+        self.present(alert, animated: true)
```
```suggestion
       present(alert, animated: true)
```

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+    }
+    
+    @objc func alertguessNumberButton() {
+        let alert = UIAlertController(title: "Cыграй в игру", message: "Введи два числа", preferredStyle: .alert)
```
убрать все ! и вынести в extention UIViewController

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+        self.present(alert, animated: true)
+    }
+
+    private func setupLayout() {
```
переделать метод

dimov-danya1999 replied:
Исправил

stacy177 commented on MainProject/MainViewController.swift:
```diff
@@ -0,0 +1,257 @@
+extension  MainViewController {
+    
+    func alertWithTF() {
+        let alert = UIAlertController(title: "Привет, Гость!", message: "Пожалуйста, введи свое ФИО", preferredStyle: .alert)
```
переделать всё

stacy177 commented on MainProject/Model/FIOModel.swift:
```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+struct FIOModel {
+    var family: String?
```
удалить опционалы

dimov-danya1999 replied:
сделал по-другому

stacy177 commented on MainProject/Model/HelloModel.swift:
```diff
@@ -0,0 +1,24 @@
+    }
+    
+    func textHello(_ text: String?) -> String {
+        if text == "leohl" {
```
переделать на тернарный оператор

dimov-danya1999 replied:
Исправил

stacy177 commented on MainProject/SceneDelegate.swift:
```diff
@@ -8,45 +8,16 @@
-        // to restore the scene back to its current state.
+        guard let windowScene = (scene as? UIWindowScene) else { return }
+        let window = UIWindow(windowScene: windowScene)
+        window.rootViewController = UINavigationController(rootViewController: MainViewController())
```
удалить NavigationController

dimov-danya1999 replied:
оставил, он нужен

dimov-danya1999 committed:
Домашняя работа с алертами. Выполнено:
1. При попадании на этот экран запрашивать ФИО через алер и отображает в UILabel
2. По нажатию на "сложение" открывать Алерт который просит ввести число 1 и число 2 и кнопка посчитать выводит результат в любом виде
3. При нажатии на кнопку "угадай число" - появляеется рандомное число
4. Создана кнопка "начать" которая вызывает Алерт с сообщением "введите слово"
5. При вводе "leohl" в Алерт  мы отдаем контролеру а контролер отдаст это в модель которая из буквосочетания "leohl" вернет слово "hello" контролеру а контролер отдаст это лейблу и отобразит это на экране

dimov-danya1999 committed:
Поправил изменения
